### ISSUE TITLE
OSX test fail: segfault during t/use.t

### ISSUE THREAD

**Issue by LLFourn** [niner/Inline-Perl5#47]
```
llfourn$ panda install Inline::Perl5
==> Inline::Perl5 depends on File::Temp
==> File::Temp depends on File::Directory::Tree
==> Fetching File::Directory::Tree
==> Building File::Directory::Tree
==> Testing File::Directory::Tree
t/basic.t .. ok
All tests successful.
Files=1, Tests=7,  0 wallclock secs ( 0.02 usr  0.01 sys +  0.34 cusr  0.03 csys =  0.40 CPU)
Result: PASS
==> Installing File::Directory::Tree
Copying blib/lib/File/Directory/Tree.pm to /Users/llfourn/.rakudobrew/moar-nom/install/share/perl6/site/lib/File/Directory/Tree.pm
==> Successfully installed File::Directory::Tree
==> Fetching File::Temp
==> Building File::Temp
==> Testing File::Temp
t/tempfile.t .. ok
All tests successful.
Files=1, Tests=14,  1 wallclock secs ( 0.01 usr  0.00 sys +  0.57 cusr  0.04 csys =  0.62 CPU)
Result: PASS
==> Installing File::Temp
Copying blib/lib/File/Temp.pm to /Users/llfourn/.rakudobrew/moar-nom/install/share/perl6/site/lib/File/Temp.pm
==> Successfully installed File::Temp
==> Fetching Inline::Perl5
==> Building Inline::Perl5
gcc -Wall p5helper.c `perl -MExtUtils::Embed -e ccopts -e ldopts` -shared -o /Users/llfourn/.panda-work/1446536082_3/blib/lib/Inline/p5helper.dylib -fPIC -g
==> Testing Inline::Perl5
t/autoload.t .................. ok
t/call.t ...................... ok
t/call_back.t ................. ok
t/call_hash.t ................. ok
t/call_p6_function_from_p5.t .. ok
t/callables.t ................. ok
t/can.t ....................... ok
t/eval.t ...................... ok
t/eval_p6_in_p5.t ............. ok
t/eval_return_values.t ........ ok
t/exceptions.t ................ ok
t/file_handles.t .............. ok
t/from.t ...................... ok
t/import.t .................... ok
t/inherit.t ................... ok
t/invoke.t .................... ok
t/invoke_p6_method_from_p5.t .. ok
t/overload.t .................. ok
t/p5_to_p6.t .................. ok
t/p6_object_destructor.t ...... ok
t/p6_to_p5.t .................. ok
t/perl5parent.t ............... ok
t/precomp.t ................... ok
t/stringify.t ................. ok
t/stringify.t ................. ok
All 4 subtests passed 
t/v6.t ........................ ok

Test Summary Report
-------------------
t/use.t                     (Wstat: 11 Tests: 4 Failed: 0)
  Non-zero wait status: 11
Files=26, Tests=345, 92 wallclock secs ( 0.07 usr  0.04 sys + 88.20 cusr  2.56 csys = 90.87 CPU)
Result: FAIL
test stage failed for Inline::Perl5: Tests failed
  in method throw at /Users/llfourn/.rakudobrew/moar-nom/install/share/perl6/runtime/CORE.setting.moarvm:1
  in method install at /Users/llfourn/.rakudobrew/moar-nom/install/share/perl6/site/lib/Panda.pm:142
  in method resolve at /Users/llfourn/.rakudobrew/moar-nom/install/share/perl6/site/lib/Panda.pm:219
  in sub MAIN at /Users/llfourn/.rakudobrew/bin/../moar-nom/install/share/perl6/site/bin/panda:18
  in block <unit> at /Users/llfourn/.rakudobrew/bin/../moar-nom/install/share/perl6/site/bin/panda:145


Failure Summary
----------------
Inline::Perl5(
	*test stage failed for Inline::Perl5: Tests failed)
```
then in panda look:

```
$ perl6 configure.pl6 
gcc -Wall p5helper.c `perl -MExtUtils::Embed -e ccopts -e ldopts` -shared -o lib/Inline/p5helper.dylib -fPIC -g
```

```
$ perl6 -Ilib t/use.t 
1..4
ok 1 - use loaded the module
ok 2 - package functions work
ok 3 - constructor works
ok 4 - package methods work
Segmentation fault: 11
```

```
perl6 -v
This is perl6 version 2015.10-141-gc0fa4af built on MoarVM version 2015.10-42-g73c0269
```

**Comment by LLFourn** [comment]
I did some digging:
```perl6
@p5.push: Inline::Perl5.new xx 10
````
Is the line that segfaults. And if replace it like:
```perl6
#@p5.push: Inline::Perl5.new xx 10;
Inline::Perl5.new; 
```
It also segfaults.

**Comment by lizmat** [comment]
Perhaps d192f1c29a6297b2607e51 is to blame?

**Comment by niner** [comment]
I think xx is innocent here since a plain Inline::Perl5.new does it as well.

Is your perl compiled with -DMULTIPLICITY? perl -V | grep MULTIPLICITY
If not, then that may be the reason. Inline::Perl5 should not allow creating multiple Inline::Perl5 objects (and thus interpreters) when perl does not support that.

**Comment by LLFourn** [comment]
My perl is not compiled with -DMULTIPLICITY :)

**Comment by flussence** [comment]
Same segfault here on Gentoo, and it goes away if I enable ithreads.

**Comment by niner** [comment]
This was fixed in commit 81243ea6518267be718bf5e092e9c09b9a36f281
We now throw a X::Inline::Perl5::NoMultiplicity instead of impolitely segfaulting. The test was adjusted to handle this exception.
